src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -395,7 +396,97 @@ struct KVAxesPosition {
     uint32_t batch;
     uint32_t seq_len;
 };
+} // anonymous namespace
 
+class CutTailAfterSlice : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("npuw::patterns::CutTailAfterSlice");
+    CutTailAfterSlice::CutTailAfterSlice(
+        std::shared_ptr<std::pair<ov::OutputVector, ov::ParameterVector>> separated_tail_protocol) {
```

Comment:
shared_ptr on pair makes no sense, we use reference_wrappers. Also, please replace the pair with a normal struct, let's stop this `.first` / `.second` abuse.